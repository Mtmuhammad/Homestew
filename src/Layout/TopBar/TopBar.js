import "./TopBar.scss";
import React from "react"
import { Link } from "react-router-dom";

const TopBar = () => {
  return (
    <>
      <div className="topbar">
        <div className="container d-flex justify-content-between">
          {/* social icons */}
          <div style={{ marginLeft: "15px" }} className="socials">
            <a data-testid="social-icon" href="#0">
              <i className="fa-brands fa-twitter"></i>
            </a>
            <a data-testid="social-icon" href="#0">
              <i className="fa-solid fa-rss"></i>
            </a>
            <a data-testid="social-icon" href="#0">
              <i className="fa-brands fa-facebook"></i>
            </a>
            <a data-testid="social-icon" href="#0">
              <i className="fa-brands fa-pinterest"></i>
            </a>
            <a data-testid="social-icon" href="#0">
              <i className="fa-brands fa-instagram"></i>
            </a>
            <a data-testid="social-icon" href="#0">
              <i className="fa-brands fa-youtube"></i>
            </a>
          </div>
          {/* end social icons */}

          {/* user icons */}
          <div className="socials">
            <Link data-testid="user-icon" to="search">
              <i className="fa-solid fa-magnifying-glass"></i>
            </Link>
          </div>
          {/* end user icons */}
        </div>
      </div>
    </>
  );
};

export default TopBar;
