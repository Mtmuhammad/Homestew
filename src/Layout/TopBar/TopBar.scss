@use "../../Styles/abstracts/mixins" as *;
@use "../../Styles/abstracts/variables" as *;

.topbar {
  background-color: $primary-green;
  height: auto;
  padding: 10px 0;

  & .container {
    max-width: 1100px;
  }
}

.socials {
  & a,
  div {
    text-decoration: none;
    color: white;
    margin-right: 20px;
    transition: all 0.3s;

    @include respond(sphone) {
      margin-right: 15px;
    }

    @include respond(xsphone) {
      margin-right: 13px;
    }
    &:hover {
      color: $icon-hover-grey;
    }

    & i {
      font-size: 2.5rem;
    }
  }
}
