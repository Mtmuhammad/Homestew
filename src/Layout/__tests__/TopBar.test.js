import React from "react";
import { render, screen } from "@testing-library/react";
import TopBar from "../TopBar/TopBar";
import {BrowserRouter } from "react-router-dom"

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<TopBar />, {wrapper: BrowserRouter});
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<TopBar />, {wrapper: BrowserRouter});
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("render all link icons", () => {
  test("should render all social icons", () => {
    render(<TopBar />, {wrapper: BrowserRouter});
    const socialIcons = screen.getAllByTestId("social-icon");
    expect(socialIcons.length).toBe(6);

    const userIcons = screen.getAllByTestId("user-icon");
    expect(userIcons.length).toBe(1);
  });
});

