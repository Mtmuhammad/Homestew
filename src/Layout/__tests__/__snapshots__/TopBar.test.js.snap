// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="topbar"
  >
    <div
      class="container d-flex justify-content-between"
    >
      <div
        class="socials"
        style="margin-left: 15px;"
      >
        <a
          data-testid="social-icon"
          href="#0"
        >
          <i
            class="fa-brands fa-twitter"
          />
        </a>
        <a
          data-testid="social-icon"
          href="#0"
        >
          <i
            class="fa-solid fa-rss"
          />
        </a>
        <a
          data-testid="social-icon"
          href="#0"
        >
          <i
            class="fa-brands fa-facebook"
          />
        </a>
        <a
          data-testid="social-icon"
          href="#0"
        >
          <i
            class="fa-brands fa-pinterest"
          />
        </a>
        <a
          data-testid="social-icon"
          href="#0"
        >
          <i
            class="fa-brands fa-instagram"
          />
        </a>
        <a
          data-testid="social-icon"
          href="#0"
        >
          <i
            class="fa-brands fa-youtube"
          />
        </a>
      </div>
      <div
        class="socials"
      >
        <a
          data-testid="user-icon"
          href="/search"
        >
          <i
            class="fa-solid fa-magnifying-glass"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
